### PR TITLE
Add Mv module

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

File: src/Streamly/Coreutils/Mv.hs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+data Mv = Force | CreateOnly
+
+force :: Mv
+force = Force
+
+createOnly :: Mv
+createOnly = CreateOnly
+
+instance Default Mv where
+    def = Force
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Mv.hs:33]: This is a very bad default.

File: src/Streamly/Coreutils/Mv.hs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import Data.Default.Class (Default(..))
+import System.Directory
+    ( renameFile
+    , renameDirectory
+    )
+
+data Mv = Force | CreateOnly
+
+force :: Mv
+force = Force
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Mv.hs:27]: `force` should be a modifier of options i.e. `force :: Switch -> Mv -> Mv`. default should be `Off`.

File: src/Streamly/Coreutils/Mv.hs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+createOnly :: Mv
+createOnly = CreateOnly
+
+instance Default Mv where
+    def = Force
+
+mv :: Mv -> FilePath -> FilePath -> IO ()
+mv options = do
+    case options of
+        Force -> renameDirectory
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Mv.hs:38]: We cannot force move a file?

File: src/Streamly/Coreutils/Mv.hs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        CreateOnly -> renameFile
```
Comment (harendra-kumar) [inline src/Streamly/Coreutils/Mv.hs:39]: We cannot move a directory?

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: src/Streamly/Coreutils/Mv.hs (hunk 4 above)
Comment (rnjtranjan) [reply src/Streamly/Coreutils/Mv.hs]: Fixed.

File: src/Streamly/Coreutils/Mv.hs (hunk 3 above)
Comment (rnjtranjan) [reply src/Streamly/Coreutils/Mv.hs]: Fixed.

**Pull request merged**